### PR TITLE
ERR: drop function code macro generation

### PR DESCRIPTION
**Pull request by levitte**
While working on #13390, I figured that we could as well complete the work by dropping the generation of `_F_` macros.  The new ones that were generated in master have been questioned more than once, since they aren't useful anywhere.

This PR includes the commits from #13390, and is pending that one being merged.

### CODE REVIEW

**Review by paulidale — COMMENTED** [review]
CIs are relevant

File: include/crypto/asyncerr.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+#ifndef OSSL_CRYPTO_ASYNCERR_H
+# define OSSL_CRYPTO_ASYNCERR_H
+# pragma once
+
+# include <openssl/opensslconf.h>
+# include <openssl/symhacks.h>
+
+# ifdef  __cplusplus
+extern "C" {
```
Comment (FdaSilvaYY) [inline include/crypto/asyncerr.h:19]: where is the closing brace ? ;)

Comment (levitte) [reply include/crypto/asyncerr.h:19]: Fixed in #13390

**Comment by levitte** [comment]
Rebased for GH action fixups

**Comment by levitte** [comment]
Rebased on fresher master

**Comment by richsalz** [comment]
>I figured that we could as well complete the work by dropping the generation of _F_ macros.

Does that have downstream "just recompile" issues?  Yes, obviously, but are they ones you care about?

**Comment by levitte** [comment]
Hmmmm, that's a good question.  Let me have another look

**Comment by levitte** [comment]
@richsalz, the thing you're missing is that I've *moved* all the `_F_` macros that we have in 1.1.1 to `include/openssl/cryptoerr_legacy.h` and `include/openssl/sslerr_legacy.h`.  This allows us to clean the other err headers.

So no, downstream recompilation should not be a problem.

**Review by t8m — APPROVED** [review]
LGTM

**Comment by richsalz** [comment]
> @richsalz, the thing you're missing is that I've _moved_ all the `_F_` macros that we have in 1.1.1 to `include/openssl/cryptoerr_legacy.h` and `include/openssl/sslerr_legacy.h`. This allows us to clean the other err headers.

Got it, makes sense, thanks.

And I guess you originally forgot that, too since you had to look :)

**Comment by levitte** [comment]
> And I guess you originally forgot that, too since you had to look :)

Yup.  I did this work quite a while ago, and my personal memory banks have their limits :wink:

**Comment by levitte** [comment]
Merged

b14c8465c0899c3c2b2acc3d01472941de4c318e ERR: clean away everything related to _F_ macros from util/mkerr.pl
e337b82410a031f0ff60ebf6744b97da2a276e51 ERR: Rebuild all generated error headers and source files

**Pull request closed without merging**